cpp/src/strings/search/findall.cu

```diff
@@ -139,5 +176,14 @@ std::unique_ptr<column> findall(strings_column_view const& input,
   return detail::findall(input, prog, stream, mr);
 }
 
+std::unique_ptr<column> find_re(strings_column_view const& input,
```

Comment:
BTW, since `find_re` is now in the same source file as `find_all`, it would be better to rename this source file into something else, such as just `find.cu` or `find_single.cu` (as we have `find_multiple.cu`).

Reply:
I will consider this in a future PR. I'm adding another API here #16900 and would want to account for that as well.